fusedrug/utils/file_formats/fasta_efficient_lookup_hdf5_index.py

```diff
@@ -0,0 +1,72 @@
+import h5py
+from Bio import SeqIO
+from typing import List
+
+
+class LargeFASTAConverter:
+    def __init__(self, chunk_size: int = 100000):
+        self.chunk_size = chunk_size
+        self.total_seen = 0
+
+    def convert_large_fasta(self, input_fasta: str, output_hdf5: str) -> None:
+        """
+        Convert FASTA to HDF5 using sequence IDs as keys.
```

Comment:
Does the hdf5 will include all the information including the description?

Reply:
no, the description is discarded, I'll need to fix it. (not in this PR).
In the next PR I'll add support for it in the db version and then possibly in the following PR in the hdf5 one.